runtime-modules/bureaucracy/src/lib.rs

```diff
@@ -226,10 +309,182 @@ decl_module! {
 
             // Trigger an event
             Self::deposit_event(RawEvent::LeaderSet(member_id, role_account_id));
+        }
 
-            Ok(())
+        /// Evict the currently set lead
+        pub fn unset_lead(origin) {
+            ensure_root(origin)?;
+
+            // Ensure there is a lead set
+            let lead = <CurrentLead<T, I>>::get();
+
+            if lead.is_none() {
+                return Err(Error::CurrentLeadNotSet);
+            }
+
+            // mutation
+
+            // Update current lead
+            <CurrentLead<T, I>>::kill();
+
+            // Unwrap lead
+            if let Some(lead) = lead {
+                // Trigger event
+                Self::deposit_event(RawEvent::LeaderUnset(lead.member_id, lead.role_account_id));
+            }
+        }
+
+        /// Update the associated role account of the active worker.
+        pub fn update_worker_role_account(
```

Comment:
Why are there two actor identifiers here? The `worker_id` should resolve to a worker profile, which holds the underlying `member_id`, which then can be used to lookup whether account signing `origin` is same as membershp controller account.

Reply:
I guess perhaps I missed this in an earlier review, but there should be `Worker.member_id`, and there would then probably need to be a `ensure_member_signed_as_worker(origin, worker_id) => Result<Worker, >`.